### ISSUE TITLE
Remote Rel-Me Verification

### ISSUE THREAD

**Issue by shiribailem** [friendica/friendica#13081]
### Is the feature request related to a problem? Please describe.

Verification doesn't appear in searches, limited to no appearance in posts (I haven't seen it but I follow almost exclusively mastodon users), and trusting a server for rel=me verification is also trusting the admin because an admin could readily falsify it (or a hacker who got access to the server)

### Describe the feature you'd like

As a low priority task the server should pull the homepage URLs from profiles and check their rel=me links, that way they can readily show as verified in search results, on all posts/comments, and on profiles when using local profile views.

Also then you would only have to trust the admin of your server that the verification is valid (and if you don't trust your own admin you're kinda SOL anyways)

### Additional context

I do not know how much support ActivityPub even has for sharing profile data so I recognize this may be complicated by that.

**Comment by MrPetovan** [comment]
At the moment we do not seem to save the "Homepage URL" field in local records of remote Friendica contacts so the local `rel="me"` verification can't be done until we actually store this piece of information.

**Comment by shiribailem** [comment]
I definitely figured that would be a requirement to implementing this.

Also occurred to me to note, just for the sake of it: this should absolutely be an admin option because it does cause additional workload.

And do updates to profile details propagate at all? Or are they manual pulls? Because if they're manually there would certainly need to be a process for refreshing I guess...

**Comment by MrPetovan** [comment]
There are regular polls for remote profile details, and there's a manual process to refresh a specific contact. For the current verification, we clear the verification status when the profile field value changes, but we don't periodically re-verify.

**Comment by shiribailem** [comment]
Not an unreasonable policy, especially for workload. More I think about that part it's not the biggest of deals, but I imagine it might be worthwhile for a long cycle refresh (like once or twice a year). It'd mostly be just checking for things like abandoned/compromised accounts, changeover of ownership in organizations, etc.

**Comment by shiribailem** [comment]
Regardless, I would like to see broader trust than just same server. The vast majority of contacts you interact with are most likely to be on different servers, and on top of that verification is needed more on remote entities than local (on local you probably trust your admin to moderate impersonators, but you don't extend that trust arbitrarily to all nodes)

**Comment by MrPetovan** [comment]
Exactly, you don't extend that trust to all nodes, so impersonation on a remote node is already mitigated by the fact that you aren't trusting remote nodes as much as your own. As a result, I don't think there's a pressing need for remote `rel="me"` verification.

On the other hand, local `rel="me"` verification is worthwhile because even if you trust your node administrator, it can be a fairly large node with thousands of active accounts where local impersonation could be profitable.

**Comment by shiribailem** [comment]
I disagree on the first especially (I don't disagree that it has a use on local, just that it's not as important)

An example: if I search for ProPublica, I'm not on newsie.social, the vast majority of users aren't on there. But if I see 3 different ProPublica accounts show up, I feel like I should know off the bat which one is valid. And especially when the default behavior is to send the user to their remote profile page, most non-proficient users will trust that node's verification not understanding that they shouldn't.

Basically, not-trusting remote nodes is more of an expectation of experts. If we want the fediverse to succeed we should cover the needs and expectations of the typical people using social media.

**Comment by MrPetovan** [comment]
If we start talking about "typical people using social media", they don't have the faintest idea what the `rel="me"` verification entails and its limitations, so remote `rel="me"` still doesn't sound pressing to me.

The Twitter blue checkmark was valuable when it was attributed by a human process. It lost all its value when it started to be sold for a subscription. So I'd be weary of giving too much importance to a technical self-verification process that I'm sure can be abused in novel and creative ways.

To go back to your example, when presented with 3 different ProPublica accounts, "typical people using social media" would seek a confirmation from a human, either a friend who's already following one of these accounts, or a known journalist from the outlet that would reference the official ProPublica account, etc... An obscure technical verification may help, but it can't be the be-all-end-all of account verification.

It's definitely a nice-to-have, but I like how discreet it is at the moment and wouldn't want to display this verification status in many more places.

**Comment by shiribailem** [comment]
Not an unreasonable take, and I sincerely appreciate the time and consideration you take with these (I do hope I'm not being a best, I'm mostly intending to just throw thoughts out there to start the discussion on various potential features when I don't see an existing conversation around it).

And I guess that also gets to one of my gripes with the way I see it elsewhere anyways, what I imagine as a personal ideal isn't just a checkmark but instead to see something like "ProPublica (propublica.org ✓)" (showing the actual verified address in a bubble with the check), which I think is both more accessible (I think it clearly indicates that the profile is associated with the domain or site given, and indicates specifically what it was verified against.

But ultimately I think your take is fair and I figure this request stands as a "if some dev gets excited about it and wants to implement it"

**Comment by foss-** [comment]
Read this issue as I have been wondering about this on various occasions. This also came up in a recent conversation about the Fedilab account https://toot.fedilab.app/@apps/110459697047646010

While the con arguments are certainly valid I still see the benefit. Having rel me is a benefit not a loss and making it visible when accounts are accessed on remote instances would help raising awareness for this feature.

**Comment by MrPetovan** [comment]
What do you mean by "when accounts are accessed on remote instances"? Here's my current thoughts on this subject:
- We could do remote contact rel="me" verification but we currently do not share the required piece of information (the Homepage profile field) between servers.
- I'd prefer not to add more places where the result of the local rel="me" verification is shown.

**Comment by foss-** [comment]
https://toot.fedilab.app/@apps shows as
<img src="https://github.com/friendica/friendica/assets/5708172/eec85efc-801a-4a29-826a-1b7f643a9e5f" width="400">

When accessed on friendica it shows as
<img src="https://github.com/friendica/friendica/assets/5708172/9cc20695-58af-4169-ba27-21408aad80cf" width="400">

and when accessed on friendica via e.g. Fedilab app, it also does not show the URL. However the URL is shown when showing the profile on their  instance.

**Comment by MrPetovan** [comment]
Remote Mastodon contact rel="me" verification is another can of worms, since they perform it on their arbitrary profile fields. We aren't there yet, we just have a single field we are checking at the moment in Friendica.

**Comment by akno77** [comment]
I'd also like to know if it is possible to display the Rel-Me verification on Mastodon pages, etc. Is this technically possible at all?
